src/d/actor/d_a_obj_hami3.cpp

```diff
@@ -4,87 +4,259 @@
 //
 
 #include "d/actor/d_a_obj_hami3.h"
+#include "SSystem/SComponent/c_lib.h"
+#include "d/d_com_inf_game.h"
 #include "m_Do/m_Do_ext.h"
 #include "d/d_procname.h"
 #include "d/d_priority.h"
+#include "m_Do/m_Do_mtx.h"
+#include "d/res/res_hami3.h"
+
+const char daObjHami3::Act_c::M_arcname[] = "Hami3";
+
+#if VERSION > VERSION_DEMO
+const char daObjHami3::Act_c::M_evname[] = "ami_cam";
+#endif
+Mtx daObjHami3::Act_c::M_tmp_mtx;
+
 
 /* 00000078-0000012C       .text nodeCallBack__FP7J3DNodei */
-static BOOL nodeCallBack(J3DNode*, int) {
+static BOOL nodeCallBack(J3DNode* node, int calcTiming) {
+    if (calcTiming == J3DNodeCBCalcTiming_In) {
+        J3DJoint* joint = (J3DJoint*)node;
+        s32 jntNo = joint->getJntNo();
+        J3DModel* model = j3dSys.getModel();
+        daObjHami3::Act_c* userArea = (daObjHami3::Act_c*)model->getUserArea();
+        if (userArea != NULL) {
+            MTXCopy(model->getAnmMtx(jntNo), *calc_mtx);
+            cMtx_YrotM(*calc_mtx, userArea->field_0x2C8);
+            model->setAnmMtx(jntNo, *calc_mtx);
+            cMtx_copy(*calc_mtx, J3DSys::mCurrentMtx);
+        }
+    }
+    return TRUE;
     /* Nonmatching */
 }
 
 /* 0000012C-0000026C       .text CreateHeap__Q210daObjHami35Act_cFv */
 BOOL daObjHami3::Act_c::CreateHeap() {
+    J3DModelData * modelData = (J3DModelData *)dComIfG_getObjectRes(M_arcname, HAMI3_BDL_HAMI3);
+#if VERSION > VERSION_DEMO
+    JUT_ASSERT(0x71, modelData != NULL);
+#else
+    JUT_ASSERT(0x70, modelData != NULL);
+#endif
```

Comment:
VERSION_SELECT macro helps reduce code duplication for version differences. Also 
```suggestion
    JUT_ASSERT(VERSION_SELECT(112, 113, 113, 113), modelData != NULL);
```